.github/actions/restore-node/action.yml

```
@@ -156,28 +168,31 @@ runs:
           git add package.json yarn.lock
           ${{ inputs.keep-endo }} || rm -rf ~/endo
         fi
-      shell: bash
-      if: steps.built.outputs.cache-hit != 'true'
+
     - name: yarn build
+      if: steps.built.outputs.cache-hit != 'true'
       working-directory: ${{ inputs.path }}
+      env:
+        XSNAP_RANDOM_INIT: ${{ inputs.xsnap-random-init }}
+      shell: bash
       run: |-
         set -ex
         yarn build
         mkdir -p node_modules/.cache/agoric
         date > node_modules/.cache/agoric/yarn-built
-      shell: bash
-      if: steps.built.outputs.cache-hit != 'true'
-      env:
-        XSNAP_RANDOM_INIT: ${{ inputs.xsnap-random-init }}
+
     - name: git dirty check
       working-directory: ${{ inputs.path }}
+      shell: bash
       run: |-
         set -x
-        # In case of Endo override, ignore matching index and worktree.
-        # (First column is non-space, second column is space, followed by separator.)
-        if [ -n "$(git status --porcelain | grep -Eve '^[^ ]  '; true)" ]; then
+        # Fail if `git status` reports anything other than the following:
+        # * an untracked endo-sha.txt from above
+        # * work tree files that have been staged without further changes
+        #   (e.g., package.json or yarn.lock) as indicated by the Y position
+        #   in "XY PATH" being a space
+        if git status --porcelain | awk '$0 !~ /^[?][?] endo-sha.txt$|^. |^$/ { exit 1; }'; then
           git status
           echo "Unexpected dirty git status" 1>&2
           exit 1
         fi
```

The logic of the `if` condition is backwards here:
```suggestion
        if git status --porcelain | awk '$0 !~ /^[?][?] endo-sha.txt$|^. |^$/ { exit 1; }'; then :
        else
          git status
          echo "Unexpected dirty git status" 1>&2
          exit 1
        fi
```

---

D'oh, nice catch! I switched back to the old approach with the new grep pattern.